fix(Scripts/GruulsLair): Olm moves freely and Kiggler's chase range is 40y

v-mstrs opened the pull request:
Olm now moves freely and Kiggler's chase range is 40yards
Source: https://www.youtube.com/watch?v=Ki2zOxFaB-4
https://www.wowhead.com/tbc/spell=36152/lightning-bolt
https://wowpedia.fandom.com/wiki/Kiggler_the_Crazed
<!-- First of all, THANK YOU for your contribution. -->

<!-- How to title your Pull Request, Description, Co-Authors (Cherry Pick) and others, please see the link below -->
<!-- https://www.azerothcore.org/wiki/commit-message-guidelines -->

## Changes Proposed:
<!-- If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address. -->
This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:
<!-- If your fix has a relating issue, link it below -->
- https://github.com/chromiecraft/chromiecraft/issues/5344
- https://github.com/azerothcore/azerothcore-wotlk/issues/23243

## SOURCE:
<!-- If you can, include a source that can strengthen your claim -->
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:
<!-- Does it build without errors? Did you test in-game? What did you test? On which OS did you test? Describe any other tests performed -->
This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.


## How to Test the Changes:
<!-- Describe in a detailed step-by-step order how to test the changes -->

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. .tele gruulslair
2. Clear trash then engage boss.

## Known Issues and TODO List:
<!-- Is there anything else left to do after this PR? -->

- [ ]
- [ ]

<!-- If you intend to contribute repeatedly to our project, it is a good idea to join our discord channel. We set ranks for our contributors and give them access to special resources or knowledge: https://discord.com/invite/GyFvXpk7)
     Do not remove the instructions below about testing, they will help users to test your PR -->
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

v-mstrs committed:
Olm now moves freely

Olm now moves freely

brighton-chi commented:
I'm not sure the deleted code is correct for Kiggler, either. If I'm understanding the deleted code correctly, it provides that the boss will move to the targeted player if they are more than 25 yards away.

Kiggler was tanked by ranged in TBC specifically to avoid his arcane explosion, which has a radius of 30 yards. If he is moving to 25 yards from his target, then it is impossible for the Kiggler tank to avoid the attack.

I don't know if 25 yards is right for Krosh, either, though the situation is not as relevant for him since his AOE has a radius of only 15 yards.

v-mstrs committed:
Change kiggler chase range to 40y

His lightning bolt has a maximum range of 40

https://www.wowhead.com/wotlk/spell=36152/lightning-bolt
https://wowpedia.fandom.com/wiki/Kiggler_the_Crazed

v-mstrs committed:
Merge branch 'master' into high-king-maulgar

sudlud committed:
Merge branch 'master' into high-king-maulgar

sudlud reviewed (approved):
Approved. The Olm change fixes the static-caster bug from #23243 (the old 25y leash let him stand still whenever the tank was inside 25y), and Kiggler's 40y leash keeps the ranged tank outside the 30y Arcane Explosion, which addresses @brighton-chi's concern. Krosh is left at 25y, but since his Blast Wave is only 15y that's safe; aligning Krosh for consistency can be a future PR if wanted. Nonblocking.

sudlud committed:
Merge branch 'master' into high-king-maulgar
